### PR TITLE
Remove BUILD reference to removed files: Fix bazel build

### PR DESCRIPTION
**Pull request by apelisse**
**What this PR does / why we need it**:
Bazel build is broken because a pull-request has removed some go files, but not the BUILD references to these file. Update the go files. I've also created an issue in test-infra(https://github.com/kubernetes/test-infra/issues/4083) as this should have been detected earlier

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:
```release-note
NONE
```

### CODE REVIEW

**Comment by mtaufen** [comment]
Was about to fix this myself. Why not remove `pkg/api/errors` entirely?

File: pkg/watch/BUILD [hunk 1]
```diff
@@ -5,11 +5,6 @@ load(
     "go_library",
```
Comment (ixdy) [inline pkg/watch/BUILD]: please remove this too

**Comment by ixdy** [comment]
agree with @mtaufen re: deleting `pkg/api/errors` entirely.

**Comment by apelisse** [comment]
Come-on, I didn't look THAT much :-)

**Comment by apelisse** [comment]
Updated

**Comment by ixdy** [comment]
/lgtm

Thanks!

File: pkg/watch/BUILD [hunk 2]
```diff
@@ -1,15 +1,5 @@
 package(default_visibility = ["//visibility:public"])
```
Comment (mikedanese) [inline pkg/watch/BUILD]: remove this file and rerun gazel

Comment (mikedanese) [inline pkg/watch/BUILD]: See https://github.com/kubernetes/kubernetes/pull/50785

**Comment by apelisse** [comment]
Done

**Comment by mikedanese** [comment]
thx

/lgtm
/approve no-issue

**Comment by mikedanese** [comment]
BUILD changes, I'm manually approving. This is blocking others.

**Comment by apelisse** [comment]
Thanks, it's pretty painful ...

**Pull request merged**